File: frontend/src/types/iteration.ts
export interface Iteration {
  id: number
  issueNumber: string
  projectCode: string
  title: string
  issueUrl: string
  status: IterationStatus
  priority: IterationPriority
  developmentNotes: string
  releaseNotes: string
  flowchartPath: string
  assigneeId: number | null
  estimatedTime: string | null
  actualTime: string | null
  apiDocUrl: string
  impactScope: string
  todos: string
  createdBy: number | null
  createdAt: string
  updatedAt: string
  statusChangedAt: string | null
  completedAt: string | null
  localDirPath: string | null
  hasTodos: boolean | null
  active: boolean
}

export type IterationStatus = 'TODO' | 'IN_PROGRESS' | 'TESTING' | 'DONE'

export type IterationPriority = 'HIGH' | 'MEDIUM' | 'LOW'

export interface IterationCreateRequest {
  issueNumber: string
  projectCode: string
  title: string
  issueUrl?: string
  status: IterationStatus
  priority: IterationPriority
  developmentNotes?: string
  releaseNotes?: string
  flowchartPath?: string
  assigneeId?: number
  estimatedTime?: string
  actualTime?: string
  apiDocUrl?: string
  impactScope?: string
  todos?: string
}

export interface IterationUpdateRequest extends Partial<IterationCreateRequest> {
  id: number
}

export interface IterationQueryParams {
  page?: number
  size?: number
  sortBy?: string
  sortDir?: string
  status?: string
  priority?: string
}

export interface IterationStatusInfo {
  value: IterationStatus
  label: string
  color: string
  order: number
}

export interface IterationPriorityInfo {
  value: IterationPriority
  label: string
  color: string
}

export const IterationStatusMap: Record<IterationStatus, IterationStatusInfo> = {
  TODO: { value: 'TODO', label: '待开发', color: '#909399', order: 1 },
  IN_PROGRESS: { value: 'IN_PROGRESS', label: '开发中', color: '#409EFF', order: 2 },
  TESTING: { value: 'TESTING', label: '测试中', color: '#13C2C2', order: 3 },
  DONE: { value: 'DONE', label: '已完成', color: '#67C23A', order: 4 }
}

export const IterationPriorityMap: Record<IterationPriority, IterationPriorityInfo> = {
  HIGH: { value: 'HIGH', label: '高', color: '#F56C6C' },
  MEDIUM: { value: 'MEDIUM', label: '中', color: '#E6A23C' },
  LOW: { value: 'LOW', label: '低', color: '#67C23A' }
}

export interface IterationImportConfig {
  id: number
  baseDirPath: string
  active: boolean
  createdAt: string
  updatedAt: string
}

export interface IterationSyncHistory {
  id: number
  iterationId: number
  syncType: 'PAGE_TO_LOCAL' | 'LOCAL_TO_PAGE'
  fieldName: string
  oldValue: string | null
  newValue: string | null
  syncedAt: string
  createdAt: string
}

export interface OtherFile {
  name: string
  path: string
}

File: backend/src/main/java/com/example/mynewwork/model/entity/Iteration.java
package com.example.mynewwork.model.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "iterations")
@Data
@EqualsAndHashCode(callSuper = false)
@EntityListeners(AuditingEntityListener.class)
public class Iteration {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(name = "issue_number", nullable = false, length = 50)
    private String issueNumber;

    @NotBlank
    @Column(name = "project_code", nullable = false, length = 100)
    private String projectCode;

    @NotBlank
    @Column(nullable = false, length = 500)
    private String title;

    @Column(name = "issue_url", length = 1000)
    private String issueUrl;

    @NotBlank
    @Column(nullable = false, length = 50)
    private String status;

    @NotBlank
    @Column(nullable = false, length = 20)
    private String priority;

    @Column(name = "development_notes", columnDefinition = "LONGTEXT")
    private String developmentNotes;

    @Column(name = "release_notes", columnDefinition = "LONGTEXT")
    private String releaseNotes;

    @Column(name = "flowchart_path", length = 1000)
    private String flowchartPath;

    @Column(name = "assignee_id")
    private Long assigneeId;

    @Column(name = "estimated_time")
    private LocalDateTime estimatedTime;

    @Column(name = "actual_time")
    private LocalDateTime actualTime;

    @Column(name = "api_doc_url", length = 1000)
    private String apiDocUrl;

    @Column(name = "impact_scope", columnDefinition = "TEXT")
    private String impactScope;

    @Column(columnDefinition = "TEXT")
    private String todos;

    @Column(name = "created_by")
    private Long createdBy;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "status_changed_at")
    private LocalDateTime statusChangedAt;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    @Column(name = "local_dir_path", length = 1000)
    private String localDirPath;

    @Column(name = "has_todos")
    private Boolean hasTodos = false;

    @Column
    private Boolean active = true;
}